app/controllers/concerns/lesson_planning.rb

```diff
@@ -23,4 +24,14 @@ def current_planned_lesson(lesson)
       planned_lessons.
       find_or_initialize_by(lesson_id: lesson.id)
   end
+
+  def current_lesson_plan_missing_lesson_in_topic?(topic)
+    topic.lessons.each do |lesson|
```

Comment:
Could also use `topic.lessons.any?{ ... }` here to be more declarative/functional. Non-blocking.